# Task
Combine and export, uncleaned WFP data file
This will involve the following steps:

1.  **Load All Uncleaned WFP Data**: Iterate through the `resource_map` to load each yearly `wfp_food_prices_global_YYYY.csv` file into a list of DataFrames, without any filtering or cleaning.
2.  **Combine All Uncleaned Data**: Concatenate all the loaded yearly DataFrames into a single `raw_master_df` containing all WFP data from 2010 to 2024.
3.  **Export Combined Uncleaned Data**: Save the `raw_master_df` to a new CSV file named `wfp_food_prices_global 2010-2024.csv` in Google Drive, ensuring no index is written.
4.  **Final Task**: Confirm the successful export of the combined, uncleaned WFP data file to the specified location.

```python
# Initialize an empty list to store raw yearly DataFrames
raw_yearly_dfs = []

# Loop through the resource_map to load each yearly WFP CSV file
print("Loading all uncleaned WFP data from 2010-2024...")
for year, fname in resource_map.items():
    filepath = os.path.join(base_path, fname)
    if not os.path.exists(filepath):
        print(f"⚠️ File not found for {year}: {filepath}. Skipping.")
        continue
    
    # Load the CSV file without any filtering or cleaning
    try:
        df_raw_yearly = pd.read_csv(filepath)
        raw_yearly_dfs.append(df_raw_yearly)
        print(f"Loaded raw data for {year}: {len(df_raw_yearly):,} records.")
    except Exception as e:
        print(f"Error loading {filepath}: {e}")

# Combine all raw yearly DataFrames into a single raw_master_df
if raw_yearly_dfs:
    raw_master_df = pd.concat(raw_yearly_dfs, ignore_index=True)
    print(f"\n✅ Combined all uncleaned WFP data: {len(raw_master_df):,} total records.")
else:
    raw_master_df = pd.DataFrame()
    print("\n⚠️ No raw WFP data was loaded. raw_master_df is empty.")

# Define the output path for the combined uncleaned dataset in Google Drive
output_file_name = "wfp_food_prices_global 2010-2024.csv"
raw_output_path = f"/content/drive/MyDrive/{output_file_name}"

# Export the raw_master_df to a CSV file
if not raw_master_df.empty:
    try:
        raw_master_df.to_csv(raw_output_path, index=False)
        print(f"✅ Combined uncleaned WFP data successfully saved to: {raw_output_path}")
        print("\n--- Final Task Confirmation ---")
        print(f"The file '{output_file_name}' containing all uncleaned WFP food price data from 2010-2024 has been successfully exported to your Google Drive.")
    except Exception as e:
        print(f"❌ Error saving the combined uncleaned WFP data to Google Drive: {e}")
else:
    print(f"❌ Cannot save an empty DataFrame. The combined uncleaned WFP data file was not created.")
```

In [2]:
base_path = '/content/'
resource_map = {
    2010: 'wfp_food_prices_global_2010.csv',
    2011: 'wfp_food_prices_global_2011.csv',
    2012: 'wfp_food_prices_global_2012.csv',
    2013: 'wfp_food_prices_global_2013.csv',
    2014: 'wfp_food_prices_global_2014.csv',
    2015: 'wfp_food_prices_global_2015.csv',
    2016: 'wfp_food_prices_global_2016.csv',
    2017: 'wfp_food_prices_global_2017.csv',
    2018: 'wfp_food_prices_global_2018.csv',
    2019: 'wfp_food_prices_global_2019.csv',
    2020: 'wfp_food_prices_global_2020.csv',
    2021: 'wfp_food_prices_global_2021.csv',
    2022: 'wfp_food_prices_global_2022.csv',
    2023: 'wfp_food_prices_global_2023.csv',
    2024: 'wfp_food_prices_global_2024.csv'
}

print(f"Defined base_path: {base_path}")
print(f"Defined resource_map with {len(resource_map)} entries.")

Defined base_path: /content/
Defined resource_map with 15 entries.


## Load All Uncleaned WFP Data

### Subtask:
Iterate through the `resource_map` to load each yearly `wfp_food_prices_global_YYYY.csv` file into a list of DataFrames, without any filtering or cleaning.


**Reasoning**:
To load all yearly WFP food prices CSV files into a list of DataFrames without any filtering or cleaning, I will implement a loop that iterates through the `resource_map`, constructs file paths, checks for file existence, loads the CSVs into DataFrames, appends them to a list, and includes error handling.



In [6]:
import os
import pandas as pd

raw_yearly_dfs = []

for year, fname in resource_map.items():
    filepath = os.path.join(base_path, fname)

    if not os.path.exists(filepath):
        print(f"⚠️ File not found for {year}: {filepath}")
        continue

    try:
        df_raw_yearly = pd.read_csv(filepath)
        raw_yearly_dfs.append(df_raw_yearly)
        print(f"Loaded raw data for {year}: {len(df_raw_yearly):,} records.")
    except Exception as e:
        print(f"❌ Error loading {filepath}: {e}")

print(f"\n✅ Successfully loaded {len(raw_yearly_dfs)} raw DataFrames.")

Loaded raw data for 2010: 32,473 records.
Loaded raw data for 2011: 44,125 records.
Loaded raw data for 2012: 58,457 records.
Loaded raw data for 2013: 71,704 records.
Loaded raw data for 2014: 80,416 records.
Loaded raw data for 2015: 89,513 records.
Loaded raw data for 2016: 101,892 records.
Loaded raw data for 2017: 114,093 records.
Loaded raw data for 2018: 122,710 records.
Loaded raw data for 2019: 144,277 records.
Loaded raw data for 2020: 288,161 records.
Loaded raw data for 2021: 335,456 records.
Loaded raw data for 2022: 325,016 records.
Loaded raw data for 2023: 322,906 records.
Loaded raw data for 2024: 295,272 records.

✅ Successfully loaded 15 raw DataFrames.


# Task
Combine all loaded yearly WFP food price data from 2010 to 2024 into a single `raw_master_df` and save it as "wfp_food_prices_global 2010-2024.csv" to your Google Drive (`/content/drive/MyDrive/wfp_food_prices_global 2010-2024.csv`).

## Combine All Uncleaned Data

### Subtask:
Concatenate all the loaded yearly DataFrames into a single `raw_master_df` containing all WFP data from 2010 to 2024.


**Reasoning**:
To combine all the loaded yearly DataFrames into a single `raw_master_df`, I will check if the `raw_yearly_dfs` list is not empty and then use `pd.concat` to concatenate them. If the list is empty, an empty DataFrame will be initialized.



In [11]:
if raw_yearly_dfs:
    raw_master_df = pd.concat(raw_yearly_dfs, ignore_index=True)
    print(f"\n✅ Combined all uncleaned WFP data: {len(raw_master_df):,} total records.")
else:
    raw_master_df = pd.DataFrame()
    print("\n⚠️ No raw WFP data was loaded. raw_master_df is empty.")


✅ Combined all uncleaned WFP data: 2,426,471 total records.


**Reasoning**:
The next step is to export the combined `raw_master_df` to a CSV file in Google Drive, as specified in the subtask instructions.



In [13]:
output_file_name = "wfp_food_prices_global 2010-2024.csv"
raw_output_path = f"/content/drive/MyDrive/{output_file_name}"

# Export the raw_master_df to a CSV file
if not raw_master_df.empty:
    try:
        raw_master_df.to_csv(raw_output_path, index=False)
        print(f"✅ Combined uncleaned WFP data successfully saved to: {raw_output_path}")
    except Exception as e:
        print(f"❌ Error saving the combined uncleaned WFP data to Google Drive: {e}")
else:
    print(f"❌ Cannot save an empty DataFrame. The combined uncleaned WFP data file was not created.")

✅ Combined uncleaned WFP data successfully saved to: /content/drive/MyDrive/wfp_food_prices_global 2010-2024.csv


## Final Task

### Subtask:
Confirm the successful export of the combined, uncleaned WFP data file to the specified location.


## Summary:

### Q&A
Yes, the combined, uncleaned WFP data file was successfully exported to the specified location `/content/drive/MyDrive/wfp_food_prices_global 2010-2024.csv`.

### Data Analysis Key Findings
*   All yearly WFP food price data from 2010 to 2024 was successfully concatenated into a single `raw_master_df`.
*   The combined `raw_master_df` contains a total of 2,426,471 records.
*   The `raw_master_df` was successfully saved as a CSV file named "wfp_food_prices_global 2010-2024.csv" to the Google Drive path `/content/drive/MyDrive/wfp_food_prices_global 2010-2024.csv`.

### Insights or Next Steps
*   The raw master dataset is now available for further processing, including data cleaning, transformation, and initial exploratory data analysis.
*   Given the large number of records, it would be beneficial to perform data type optimization and check for memory efficiency before extensive analysis.
